diff -Naur Mliteraki/Mliter.cpp Mliteraki_new/Mliter.cpp
--- Mliteraki/Mliter.cpp	2004-06-17 11:55:09.477040360 +0200
+++ Mliteraki_new/Mliter.cpp	2004-06-17 11:53:06.667710216 +0200
@@ -8,7 +8,7 @@
 #define tLINUX
 //#define tDEBUG //dodatkowe komentarze
 //#define tDEBUG2 //wnikliwe komentarze
-#include <stdlib.h> //dla rand()
+#include <stdlib.h> //dla rand() i getenv()
 #include <wx/panel.h>
 #include <wx/ffile.h>
 #include <wx/string.h>
@@ -85,7 +85,7 @@
 char* plik_listwa;
 
 
-char* plik_konfig="./Mliter.kfg";
+char* plik_konfig; 
 char* plik_zmn="./graf/pzmn.JPG";
 char* plik_btnTak="./graf/btn3tak.JPG";
 char* plik_btnNie="./graf/btn3nie.JPG";
@@ -5328,6 +5328,10 @@
 		wxImageHandler *icohnd=new wxICOHandler();
 		wxImage::AddHandler( icohnd );
 
+		plik_konfig = new char[256];
+		plik_konfig = strcpy((char *)plik_konfig,(char*)getenv("HOME"));
+		plik_konfig = strcat((char *)plik_konfig,(char*)"/Mliter.kfg");
+		
 		if (wczytajPlikKonf()>0) wczytajDef(); //jesli blad przy pliku to defaulty
 		wczytajSciezkiGraf();
 		wczytajKDAWG();
